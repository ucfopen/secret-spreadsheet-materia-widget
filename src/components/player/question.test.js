jest.mock('react', () => {
	const ActualReact = jest.requireActual('react')
	return {
		...ActualReact,
		useRef: jest.fn()
	}
})

import React from 'react';
import Question from './question';
import { create, act } from 'react-test-renderer';

describe(`Question component`, () => {
	beforeEach(() => {
		jest.resetModules();
		React.useRef.mockClear();
	});

	test(`Is rendered with question body`, () => {
		const mockFocus = jest.fn();
		React.useRef.mockReturnValue({
			current: {
				focus: mockFocus
			}
		});

		const props = {
			handleQuestionToggle: jest.fn(),
			question: `Test entry question`,
			description: `Test question body`
		};

		let component
		act(() => {
			component = create(<Question {... props} />, {
				createNodeMock: () => ({
					focus: mockFocus
				})
			});
		})
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();

		// bonus test - make sure the ref works and auto-focuses
		expect(mockFocus).toHaveBeenCalled();
	});

	test(`Is rendered without question body`, () => {
		const mockFocus = jest.fn();
		React.useRef.mockReturnValue({
			current: {
				focus: mockFocus
			}
		});

		const props = {
			handleQuestionToggle: jest.fn(),
			question: `Test entry question`,
			description: ``
		};

		// not sure if this is even possible, but...
		// update the ref to null to make sure nothing calls 'focus'?
		let component
		act(() => {
			component = create(<Question {... props} />, {
				createNodeMock: () => null
			});
		})
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();

		expect(mockFocus).not.toHaveBeenCalled();
	});
});
