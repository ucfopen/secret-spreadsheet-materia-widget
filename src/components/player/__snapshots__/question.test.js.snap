// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component Is rendered with question body 1`] = `
<div
  className="popup question"
>
  <div>
    <p
      aria-label="Question: Test entry question: Test question body"
      className="mainQuestion"
      tabIndex="0"
    >
      Test entry question
    </p>
    <p>
      Test question body
    </p>
    <button
      aria-label="After closing this dialog focus will be automatically moved to the first empty cell in the spreadsheet."
      onClick={[MockFunction]}
      tabIndex="0"
      type="button"
      value="Next"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Question component Is rendered without question body 1`] = `
<div
  className="popup question"
>
  <div>
    <p
      aria-label="Question: Test entry question: "
      className="mainQuestion"
      tabIndex="0"
    >
      Test entry question
    </p>
    <button
      aria-label="After closing this dialog focus will be automatically moved to the first empty cell in the spreadsheet."
      onClick={[MockFunction]}
      tabIndex="0"
      type="button"
      value="Next"
    >
      Next
    </button>
  </div>
</div>
`;
